### PR TITLE
Correct forecast reference time hour creation in estimate-dz-rescaling

### PR DESCRIPTION
**Pull request by gavinevans**
Related to: https://github.com/metoppv/mo-blue-team/issues/563

Description
This PR makes a minor amendment to ensure that the `forecast_reference_time_hour` coordinate is always computed by subtracting the forecast period from the validity time coordinate. This is required because it is possible to provide a file with a `forecast_reference_time` coordinate with more than one forecast reference time hour, if there is a leniency allowed. This leniency is primarily desired for longer lead time forecasts where some validity time and forecast period combinations may be irregularly produced. 

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver/calibration/dz_rescaling.py [hunk 1]
```diff
@@ -219,8 +220,13 @@ def _create_hour_coord(self, source_cube: Cube, target_cube: Cube):
             target_cube: Cube to which an auxiliary coordinate will be added.
         """
         coord_name = "forecast_reference_time"
```
Comment (bayliffe) [inline improver/calibration/dz_rescaling.py:222]: This is only used once now, so could just be defined in the `long_name=` line below.

File: improver_tests/calibration/dz_rescaling/test_estimate_dz_rescaling.py [hunk 2]
```diff
@@ -246,6 +246,9 @@ def test_estimate_dz_rescaling(
     truth_adjustment is applied to the truth data to show how larger and inverted
     data (w.r.t. height) behave.
     """
+    if n_frts > 1 and len(forecast_periods) > 1:
+        return
+
```
Comment (bayliffe) [inline improver_tests/calibration/dz_rescaling/test_estimate_dz_rescaling.py:251]: Is there a catch for this in operational use, or would it simply not happen?

**Review by bayliffe — COMMENTED** [review]
Tiny comment and question.

File: improver/calibration/dz_rescaling.py (hunk 1 above)
Comment (gavinevans) [reply improver/calibration/dz_rescaling.py]: I've modified this as suggested.

File: improver_tests/calibration/dz_rescaling/test_estimate_dz_rescaling.py (hunk 2 above)
Comment (gavinevans) [reply improver_tests/calibration/dz_rescaling/test_estimate_dz_rescaling.py:251]: We don't process cubes of this shape in practice. As shown in the acceptance test data, the current set-up involves possibly multiple forecast reference times for a single lead time. We never process multiple forecast reference times and multiple forecast periods at once.

**Review by bayliffe — APPROVED** [review]

**Pull request merged**